test/com/tw/pkg/deb/repo/DebianRepositoryTest.java

```diff
@@ -4,40 +4,59 @@
 import org.apache.commons.io.FileUtils;
 import org.junit.Assert;
 import org.junit.Ignore;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
 
 import java.io.BufferedReader;
 import java.io.File;
 import java.io.FileReader;
 import java.util.ArrayList;
 import java.util.List;
 
+import static org.junit.Assert.assertEquals;
+
 public class DebianRepositoryTest {
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
     @Test
     public void shouldCheckDebainReporitoryValidityCorrectly() throws Exception {
-        DebianRepository debianRepository1 = new DebianRepository("http://in.archive.ubuntu.com/ubuntu/dists/saucy/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
-        Assert.assertEquals(true, debianRepository1.isRepositoryValid());
+        DebianRepository debianRepository1 = new DebianRepository("http://archive.ubuntu.com/ubuntu/dists/xenial/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
+        assertEquals(true, debianRepository1.isRepositoryValid());
 
-        DebianRepository debianRepository2 = new DebianRepository("http://in.archive.ubuntu.com/ubuntu/dists/invalid/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
-        Assert.assertEquals(false, debianRepository2.isRepositoryValid());
+        DebianRepository debianRepository2 = new DebianRepository("http://archive.ubuntu.com/ubuntu/dists/INVALID/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
+        assertEquals(false, debianRepository2.isRepositoryValid());
     }
 
     @Test
     public void shouldTellIfRepositoryHasChangesCorrectly() throws Exception {
-        DebianRepository debianRepository = new DebianRepository("http://in.archive.ubuntu.com/ubuntu/dists/saucy/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
+        DebianRepository debianRepository = new DebianRepository("http://archive.ubuntu.com/ubuntu/dists/xenial/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
         debianRepository.clearDownloadFolder();
         FileUtils.deleteQuietly(new File(debianRepository.getLastKnowDateStoreFilePath()));
         debianRepository.setKnownDate(0L);
-        Assert.assertEquals(true, debianRepository.hasChanges());
-        Assert.assertEquals(false, debianRepository.hasChanges());
+        assertEquals(true, debianRepository.hasChanges());
+        assertEquals(false, debianRepository.hasChanges());
     }
 
     @Ignore
     @Test
     public void shouldFetchPackageDataCorrectly() throws Exception {
-        DebianRepository debianRepository = new DebianRepository("http://in.archive.ubuntu.com/ubuntu/dists/saucy/main/binary-arm64/Packages.gz", "/tmp/getPackagesForQuery");
+        DebianRepository debianRepository = new DebianRepository("http://archive.ubuntu.com/ubuntu/dists/xenial/main/binary-amd64/Packages.gz", "/tmp/getPackagesForQuery");
```

Comment:
Also, if the plugin is doing temporary files, could we have them downloaded to `work/plugins/PLUGIN_ID` instead of whatever this plugin uses? Additionally - have an @OnUnload to remove the directory on uninstall